e3sm_diags/derivations/acme.py

```diff
@@ -147,7 +148,23 @@ def molec_convert_units(var, molar_weight):
     # Convert molec/cm2/s to kg/m2/s
     if var.units == "molec/cm2/s":
         var = var / AVOGADOR_CONS * molar_weight * 10.0
-        var.units == "kg/m2/s"
+        var.units = "kg/m2/s"
+    return var
+
+
+def a_num_sum(var):
+    # Calculate: total aerosol number concentration (#/cm3)
+    var = var * AIR_DENS / 1e6
+    var.units = "/cm3"
+    var.long_name = "aerosol number concentration"
+    return var
+
+
+def so4_mass_sum(var):
+    # Calculate: SO4 mass conc. (ng/m3) (< 1um)
+    var = var * AIR_DENS * 1e9
```

Comment:
@lannyzxj Xiaojian,this formula converts so4 mass conc. from (kg/kg) to ug/m3 which gives comparable results to models. But in your version, the results are in ug/cm3. I might get something wrong here..would you help me double check.

Reply:
Dear Dr. Zhang,

Thanks for letting me know. I have cross-compared your plots and mine offline, the plots look good.
My units are also ug/m3 for the sulfate output in my plot, which looks comparable to yours.

I double-checked my processing code for the model so4, 

it was first converted to ng/m3 after reading the data:
 "
iso4_Acc_tmp=iso4_Acc_tmp*1e12*1.225 #kg/kg to ng/kg then to ng/m3
"

and then further divided by 1000 in order to convert to ug/m3 in the data output part of the code:
"
#SO4
ivar=np.reshape(so4_all,(numday*24),order='C')
ivar=ivar/1000. #convert to ug/m3
Mvar=MV2.masked_array(ivar,mask=[np.isnan(ivar)],fill_value=-9999.)
Mvar.id="sulfate"
Mvar.long_name='Sulfate (<1um) Mass Conc.'
Mvar.units='ug/m3'
Mvar.missing_value=-9999.
Mvar.setAxis(0,hrtime_axes)
fout.write(Mvar)
"
The above processes should be equivalent to your "var = var * AIR_DENS * 1e9".

 Would you point me to where my version states the unit as ug/cm3? That could be a typo in stating the unit, the actual calculation should indeed yield ug/m3.

Best,
Xiaojian

Reply:
Thank you for your prompt reply! I missed the second units conversion step in your script when writing out files! It is good to know that our results are consistent. 
I was looking at a version of ppt (for verification), which includes aerosol diurnal cycle plots that has units in ug/cm3, which could be a typo. And I also confirmed that the ARM Diags standalone run provide the correct units!

Reply:
Good to know that! Thank you!